src/github.com/matrix-org/dendrite/syncserver/storage/output_room_events_table.go

```diff
@@ -141,8 +164,5 @@ func (s *outputRoomEventsStatements) Events(txn *sql.Tx, eventIDs []string) ([]g
 		}
 		result = append(result, ev)
 	}
-	if i != len(eventIDs) {
-		return nil, fmt.Errorf("failed to map all event IDs to events: (got %d, wanted %d)", i, len(eventIDs))
-	}
```

Comment:
This diff is misleading. What I actually did was factor out `rowsToEvents` which was being called in 2 places, and then added a 3rd call to `rowsToEvents` for the new code I added (`RecentEventsInRoom`). The diff suggests I've butchered a random function to convert it into `rowsToEvents`.